src/release/upgrade.md

```diff
@@ -19,22 +19,67 @@ $ flutter upgrade
 This command gets the most recent version of the Flutter SDK
 that's available on your current Flutter channel.
 
-If you want an even more recent version of the Flutter SDK,
-switch to a less stable Flutter channel
+If you are using the **stable** channel
+and want an even more recent version of the Flutter SDK,
+switch to the **beta** channel using `flutter channel beta`,
 and then run `flutter upgrade`.
 
+### Keeping informed
+
+We publish [migration guides] for known breaking changes.
+
+We send announcements regarding these changes to the
+[Flutter announcements mailing list][flutter-announce].
+
+To avoid being broken by future versions of Flutter,
+consider submitting your tests to our [test registry].
+
 
 ## Switching Flutter channels
 
-Flutter has three [release channels][]:
-**stable**, **beta** and **master**.
+Flutter has two release channels:
+**stable** and **beta**.
 
-{{site.alert.info}}
-  The `dev` channel was retired as of Flutter 2.8.
-{{site.alert.end}}
+### The **stable** channel
+
+We recommend the **stable** channel
+for new users
+and for production app releases.
+It is updated roughly every three months,
+and may additionally receive occasional hot fixes
+for high-severity or high-impact issues.
+
+The Flutter team's plugins and packages are continually tested against the latest **stable** release.
+
+The latest documentation for the **stable** branch is at: <https://api.flutter.dev>
+
+### The **beta** channel
+
+The **beta** channel has the latest stable release.
+This is the most recent version of Flutter that we have heavily tested.
+This channel has passed all our public testing,
+has been verified against test suites for Google products that use Flutter,
+and has been vetted against [contributed private test suites][test registry].
+The **beta** channel receives regular hot fixes
+to address newly discovered important issues.
+
+The **beta** channel is essentially the same as the **stable** channel
+but updated monthly instead of quarterly.
+Indeed, when the **stable** channel is updated,
+it is updated to the latest **beta** release.
+
+### Other channels
 
-We recommend using the **{{site.sdk.channel}}** channel
-unless you need a more recent release.
+We currently have one other channel, **master**.
+This is the channel used by people [contributing to Flutter][].
+
+This channel is not as thoroughly tested as the **beta** and **stable** channels.
+
+We do not recommend using this channel as it more likely to contain serious regressions.
+
+The latest documentation for the **master** branch is at: <https://master-api.flutter.dev>
```

Comment:
**issues:** Write in active voice per [GSG](https://developers.google.com/style/voice).
```suggestion
We have one other channel, **master**.
[Flutter contributors][] use this channel.

We test this channel less than the **beta** and **stable** channels.

We do not recommend using this channel as it might contain serious regressions.

To learn about the latest API documentation for the **master** branch,
check out <https://master-api.flutter.dev>
```

Reply:
I don't think "Flutter contributors" is a good link text to use for linking to the contributing guide. The guide doesn't list the people contributing. It explains how to contribute to Flutter.

I don't think "To learn about the latest API documentation" makes sense, after all, that link doesn't teach you anything about the API documentation, it _is_ the API documentation.

Reply:
i'm also a bit reluctant to say "we" test the channels, given that there are a number of different processes for testing that aren't all "we" (for one, most of the testing is automated).